js/imageeditor/shared/utils/parse_placeholder.ts

```diff
@@ -0,0 +1,21 @@
+export function parse_placeholder(text: string): string {
+	if (!text) return "";
+
+	const lines = text.split("\n");
+
+	const processed_lines = lines.map((line) => {
+		line = line.trim();
+
+		if (line.startsWith("# ")) {
+			return `<h1>${line.slice(2)}</h1>`;
+		}
+
+		if (line) {
+			return `<p>${line}</p>`;
+		}
+
+		return "";
+	});
+
+	return processed_lines.join("");
+}
```

Comment:
Instead of this, I think we can just extract any headings and preserve the whitespace in the paragraph to avoid DOM injections.

I made a REPL cos that was easier: https://svelte.dev/repl/fb4577b0993a47579a981af6be3d6b36?version=4.2.18